packages/Operators/test/tstReferenceHexShapeFunction.cpp

```diff
@@ -0,0 +1,164 @@
+/*!
+ * \file   tstReferenceHexShapeFunction.cpp
+ * \author Stuart Slattery
+ * \date   Wed May 25 12:36:14 2011
+ * \brief  Nodal shape function test.
+ */
+//---------------------------------------------------------------------------//
+
+#include "reference_implementation/DTK_ReferenceHexShapeFunction.hpp"
+#include "reference_implementation/DTK_ReferenceHex.hpp"
+#include "reference_implementation/DTK_ReferenceNode.hpp"
+
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Teuchos_Array.hpp>
+
+//---------------------------------------------------------------------------//
+// Hex test.
+TEUCHOS_UNIT_TEST( ReferenceHexShapeFunction, hex_test )
+{
+    // Create the nodes;
+    int num_nodes = 8;
+    Teuchos::Array<DataTransferKit::Entity> nodes( num_nodes );
+    nodes[0] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 0, 0.0, 0.0, 0.0 );
+    nodes[1] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 1, 2.0, 0.0, 0.0 );
+    nodes[2] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 2, 2.0, 2.0, 0.0 );
+    nodes[3] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 3, 0.0, 2.0, 0.0 );
+    nodes[4] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 4, 0.0, 0.0, 2.0 );
+    nodes[5] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 5, 2.0, 0.0, 2.0 );
+    nodes[6] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 6, 2.0, 2.0, 2.0 );
+    nodes[7] =
+        DataTransferKit::UnitTest::ReferenceNode( 0, 7, 0.0, 2.0, 2.0 );
+    
+    // Make a hex.
+    DataTransferKit::Entity hex = 
+        DataTransferKit::UnitTest::ReferenceHex( 0, 0, nodes );
+    
+    // Create a shape function.
+    Teuchos::RCP<DataTransferKit::EntityShapeFunction> shape_function =
+	Teuchos::rcp( new DataTransferKit::UnitTest::ReferenceHexShapeFunction() );
+
+    // Test the shape function dof ids for the hex.
+    Teuchos::Array<DataTransferKit::SupportId> dof_ids;
+    shape_function->entitySupportIds( hex, dof_ids );
+    TEST_EQUALITY( num_nodes, dof_ids.size() );
+    for ( unsigned n = 0; n < num_nodes; ++n )
+    {
+	TEST_EQUALITY( dof_ids[n], nodes[n].id() );
+    }
+
+    // Test the value evaluation for the hex.
+    Teuchos::Array<double> ref_point( 3, 0.0 );
+    Teuchos::Array<double> values;
+    shape_function->evaluateValue( hex, ref_point(), values );
+    TEST_EQUALITY( values.size(), num_nodes );
+    for ( unsigned n = 0; n < num_nodes; ++n )
+    {
+	TEST_EQUALITY( values[n], 1.0 / num_nodes );
+    }
+    ref_point[0] = -1.0;
+    ref_point[1] = -1.0;
+    ref_point[2] = -1.0;
+    shape_function->evaluateValue( hex, ref_point(), values );
+    TEST_EQUALITY( values.size(), num_nodes );
+    TEST_EQUALITY( values[0], 1.0 );
+    for ( unsigned n = 1; n < num_nodes; ++n )
+    {
+	TEST_EQUALITY( values[n], 0.0 );
+    }    
+
+    // Test the gradient evaluation for the hex.
+    Teuchos::Array<Teuchos::Array<double> > grads;
+    ref_point.assign( 3, 0.0 );
```

Comment:
Add comment to say you will evaluate shape function values/gradients in the center of the reference cell.